apps/checkin/views.py

```diff
@@ -0,0 +1,124 @@
+
+    def post(self, request):
+        if request.POST:
+            try:
+                uuid = request.POST["id"]
+            except KeyError:
+                return error_response("Must include id of the CheckinGroup.",
+                                      "You must include the id from the bar code in the request body.",
+                                      400)
+            try:
+                group = CheckInGroup.objects.get(id=uuid)
+            except CheckInGroup.DoesNotExist:
+                return error_response("Must include id of the CheckinGroup.",
+                                      "The id you specified does not exist in the database.",
+                                      404)
+            if group.checked_in:
+                return error_response("User already checked in.",
+                                      "The user specified is already checked in.",
+                                      409)
+            group.checked_in = True
+            group.save()
+            event = CheckInEvent(check_in_group=group, check_in=True, time=timezone.now())
+            event.save()
+            return success_data_jsonify()
+        else:
+            return error_response("Invalid method.", "Please use a post request.", 405)
+        # TODO: will need to be authenticated to admins
+
+
+class CheckOut(APIView):
+    permission_classes = (IsAdminUser,)
+
+    def post(self, request):
+        if request.POST:
+            if request.POST:
+                try:
+                    uuid = request.POST["id"]
+                except KeyError:
+                    return error_response("Must include the CheckInGroup's id.",
+                                          "You must include the user's id from the bar code in the request body.",
+                                          400)
+                try:
+                    group = CheckInGroup.objects.get(id=uuid)
+                except CheckInGroup.DoesNotExist:
+                    return error_response("Must include the CheckInGroup's id.",
+                                          "The id you specified does not exist in the database.",
+                                          404)
+                if not group.checked_in:
+                    return error_response("User is not checked in.",
+                                          "A user must be checked in to be checked out.",
+                                          409)
+                event = CheckInEvent(check_in_group=group, check_in=False, time=timezone.now())
+                group.checked_in = False
+                event.save()
+                group.save()
+                return success_data_jsonify()
+            else:
+                return error_response("Invalid method.", "Please use a post request.", 405)
+            # TODO: will need to be authenticated to admins
+
+
+# TODO: move the below functions to utils somewhere so everyone can use them
+def success_data_jsonify( code=200):
```

Comment:
i'd prefer if you just nuke this method and use Response directly, but up to you